syntax = "proto3";

package risk_engine;

option java_package = "com.titan.riskengine";
option java_multiple_files = true;

service RiskEngineService {
  rpc CheckRisk (RiskCheckRequest) returns (RiskCheckResponse);
}

message RiskCheckRequest {
  string user_id = 1;
  double amount = 2;
}

message RiskCheckResponse {
  int32 risk_score = 1;
  string risk_level = 2; 
  string action = 3;     
}